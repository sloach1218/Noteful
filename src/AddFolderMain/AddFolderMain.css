label, input, select{
  display:block;
  font-size: 14px;;
}
legend{
  font-size:20px;
  padding:0;
}
form{
  padding:10px 20px;
}