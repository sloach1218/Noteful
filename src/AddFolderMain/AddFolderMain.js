import React from 'react';
import NotesContext from '../NotesContext';
import ValidationError from '../ValidationError';

class AddFolderMain extends React.Component {

  static contextType = NotesContext;
  constructor(props) {
    super(props);
    this.state = {
      name: {
        value: "",
        touched: false
      },
      
    };
  }
  updateName(name) {
    this.setState({ name: { value: name, touched: true } });
  }
  handleSubmit(event) {
    event.preventDefault();
    const { name } = this.state;
    const data = {"name": name.value}

    fetch('http://localhost:9090/folders', {
      method: 'POST',
      headers: {
        'content-type': 'application/json'
      },
      body: JSON.stringify(data),
      })
      .then(() => {
        window.location.href = '/';
      }).catch((err) => {
        alert(`Something went wrong: ` + err + ` Please try again later.`);
      });
    
  }

  validateName() {
    const name = this.state.name.value.trim();
    if (name.length === 0) {
      return "Name is required";
    } else if (name.length < 2) {
      return "Name must be at least 2 characters long";
    }
  }
 

  render(){
    
    
    return (
      <form onSubmit={e => this.handleSubmit(e)}>
        <legend>Create a folder</legend>
        <label htmlFor="name">Folder Name:</label>
        <input 
          type="text" 
          name="name" 
          id="name" 
          onChange={e => this.updateName(e.target.value)}
          aria-label="New folder name"
          aria-required="true"></input>
        {this.state.name.touched && (<ValidationError message={this.validateName()} />)}
        <button type="submit" disabled={ this.validateName() }>Submit</button>
      </form>
    );
  }
}

export default AddFolderMain;