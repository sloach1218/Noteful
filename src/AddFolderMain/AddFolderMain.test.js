import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import AddFolderMain from './AddFolderMain';

