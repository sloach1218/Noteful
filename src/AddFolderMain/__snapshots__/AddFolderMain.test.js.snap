// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<form
  onSubmit={[Function]}
>
  <legend>
    Create a folder
  </legend>
  <label
    htmlFor="name"
  >
    Folder Name:
  </label>
  <input
    id="name"
    name="name"
    onChange={[Function]}
    type="text"
  />
  <button
    disabled="Name is required"
    type="submit"
  >
    Submit
  </button>
</form>
`;
